source/DistanceMap.cpp

```diff
@@ -173,7 +173,8 @@ bool DistanceMap::Edge::operator<(const Edge &other) const
 // source system or the maximum count is reached.
 void DistanceMap::Init(const Ship *ship)
 {
-	if(!center)
+	if(!center || (ship && !ship->GetPersonality().IsUnrestricted() &&
+			ship->GetGovernment()->IsRestrictedFrom(*center)))
```

Comment:
@warp-core noted that making NPCs follow travel restrictions means that if we add restrictions to a government and and make NPCs unrestricted in an existing mission, any old saves that already have that mission (and therefore don't have the unrestricted personality) will follow the restrictions and potentially cause missions to become unbeatable.

As such, the default should be that NPCs don't follow travel restrictions, and we have a personality to have them follow the restrictions.

Also, for the sake of making this code cleaner (and better supporting future changes like #9299), perhaps we should have Ship::IsRestrictedFrom as a function.

```c++
bool Ship::IsRestrictedFrom(const Planet &planet)
{
	// The player's ships have no travel restrictions.
	if(isYours)
		return false;

	bool restrictedByGov = government && government->IsRestrictedFrom(*planet)
	// Special ships (such as NPCs) are unrestricted by default and must be explicitly restricted
	// by their government's travel restrictions in order to follow them.
	if(isSpecial)
		return personality.IsRestricted() && restrictedByGov;
	return !personality.IsUnrestricted() && restrictedByGov;
}
```

Reply:
alr, done and I added a personality for it